### PR TITLE
bach files support installation path containing spaces

### PR DESCRIPTION
**Pull request by michelou**
Small follow-up of PR #13006.
Related to [issue 12502](https://github.com/lampepfl/dotty/issues/12502) (***WIP*** : MSI installer created using [WiX](https://wixtoolset.org/)).

The changes are Windows-specific **and** have no impact on the CI build.

In particular, [`scalac.bat`](https://github.com/lampepfl/dotty/blob/master/dist/bin/scalac.bat) should also work correctly when the Scala 3 installation path contains spaces (e.g. `C:\Program Files\Scala 3\`).

### CODE REVIEW

File: dist/bin/scalac.bat [hunk 1]
```diff
@@ -146,21 +146,21 @@ goto :eof
...
 
 @rem # jline
-set __TOOLCHAIN=%__TOOLCHAIN%%_JLINE_READER%%_PSEP%
-set __TOOLCHAIN=%__TOOLCHAIN%%_JLINE_TERMINAL%%_PSEP%
-set __TOOLCHAIN=%__TOOLCHAIN%%_JLINE_TERMINAL_JNA%%_PSEP%
-set __TOOLCHAIN=%__TOOLCHAIN%%_JNA%
+set "__TOOLCHAIN=%__TOOLCHAIN%%_JLINE_READER%%_PSEP%"
+set "__TOOLCHAIN=%__TOOLCHAIN%%_JLINE_TERMINAL%%_PSEP%"
+set "__TOOLCHAIN=%__TOOLCHAIN%%_JLINE_TERMINAL_JNA%%_PSEP%"
+set "__TOOLCHAIN=%__TOOLCHAIN%%_JNA%
```
Comment (anatoliykmetyuk) [inline dist/bin/scalac.bat:163]: No closing quote here

Comment (michelou) [reply dist/bin/scalac.bat]: Good catch ! Adressed review.

**Review by liufengyun — APPROVED** [review]

**Review by anatoliykmetyuk — APPROVED** [review]

**Comment by michelou** [comment]
@anatoliykmetyuk
**Backport candidate** to [`release-3.1.0`](https://github.com/lampepfl/dotty/tree/release-3.1.0) branch ?!

**Comment by anatoliykmetyuk** [comment]
@michelou #13634

**Comment by michelou** [comment]
@anatoliykmetyuk Thanks !

**Pull request merged**